.ci/README.md

```diff
@@ -1,63 +1,87 @@
 Concourse CI tools for MagicModules and Google Providers
 ===
 
-These tools manage the downstream repositories of [magic-modules](https://github.com/GoogleCloudPlatform/magic-modules).
+These tools manage the downstream repositories of [magic-modules](https://github.com/GoogleCloudPlatform/magic-modules), and are collectively referred to as "The Magician".
 
-# Jobs
-The concourse pipeline defined here runs through four stages when a Github pull request is opened against `magic-modules`.
-## `mm-generate`
-The first job's goal is to generate the downstream repositories of MagicModules.  This is done almost entirely by the `generate` task.
+# CI For Downstream Developers
+If you're interested in developing the repositories that manage MagicModules, here are the things you'll want to know.
 
-### `generate`
-`generate` takes two inputs:
-* The magic-modules repository, with the pull request's `head` checked out and the repo in "detached `HEAD`" state.
-* This CI repository.
+## What The Magician Does
+The Magician takes the PR you write against MagicModules and creates the downstream (generated) commits that are a result of your changes.  It pushes the downstream commits to the relevant downstream repositories, and opens PRs against them, referencing your open PR against MagicModules.  When those PRs are merged, it updates the submodules under `build/` and merges your MagicModules PR.
 
-It then runs `magic-modules/generate.yml`, which specifies one output:
+## Your Workflow
 
-* The magic-modules repository, after generation has been accomplished.
+You'll write some commits, likely modifying files under `provider` or `products`, and push a PR.  Since The Magician needs to be able to add commits to your branch to update submodules, **make sure the branch of your PR is in `GoogleCloudPlatform/magic-modules`**.
 
-After that, the generated repositories are uploaded to GitHub, in the concourse process runner's forks.  The pull request is updated to point to those forks.
+The Magician will take over, running the generator and running the downstream tests.  If the tests fail, you'll see "check failed" status on your PR<!--TODO(@ndmckinley) - better reporting and logs of test failures.-->.  When all the tests pass, the PR or PRs created by The Magician will link back to your MagicModules PR.  From there, it is your responsibility to engage with *all* the downstream PRs - The Magician doesn't test or merge those unless otherwise indicated.  You cannot merge your MagicModules PR until *all* downstreams accept the changes.  If a downstream requests changes and you need to modify your MagicModules PR, simply push another commit to your MagicModules PR and The Magician will push any generated updates to the downstream PRs.  Note that if you merge one PR, then another downstream asks for changes which affect the already-merged PR, you will have a messy situation - use caution when many repositories are impacted by your change.
```

Comment:
> Note that if you merge one PR, then another downstream asks for changes which affect the already-merged PR, you will have a messy situation - use caution when many repositories are impacted by your change

I thought the Magician would open a new PR in the repo where the PR was already merged... If that is not the case, what is the behavior?

Reply:
That's what is *supposed* to happen - I just haven't tested it and won't really be able to until we have two repos here.  :)

Reply:
I'm now sure that what will happen is that it will open a new PR *if* you have deleted the branch for the old one... but it's messy nonetheless.